src/cir.rs

```diff
@@ -63,10 +56,27 @@ pub trait Translate {
 impl Translate for Expression {
     type Output = CirProgram;
     fn translate(&self) -> Self::Output {
-        let mut cir = CirProgram::new();
-        cir_translate_helper(self, &mut cir);
-        cir.instructions.shrink_to_fit(); // shrink the memory
-        cir
+        let mut cir_instructions = Vec::new();
+        cir_translate_helper(self, &mut cir_instructions);
+        cir_instructions.shrink_to_fit();
```

Comment:
```suggestion
```

The `Vec::into_boxed_slice()` also discards excess capacity.

Reply:
committed.